### PR TITLE
Folder name with trailing space fix

### PR DESCRIPTION
**Pull request by Outrider0x400**
Origin Issue #18457 

It reminds me of #16735. So I actually added the trailing space fix for that folder creation as well.

### CODE REVIEW

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -250,8 +250,12 @@ class ProjectDialog : public ConfirmationDialog {
 
 	void _create_folder() {
 
-		if (project_name->get_text() == "" || created_folder_path != "")
+		if (project_name->get_text() == "" || created_folder_path != "" || project_name->get_text().ends_with(".") || project_name->get_text().ends_with(" "))
+		{
```
Comment (akien-mga) [inline editor/project_manager.cpp]: As per our style guide the bracket should be on the same line as the `if`.

Comment (Falke117) [reply editor/project_manager.cpp]: I'll fix it later today.

Comment (Falke117) [reply editor/project_manager.cpp]: Sorry for the delay. Fixed now with commit#164dcea

**Comment by akien-mga** [comment]
Thanks!

**Comment by hpvb** [comment]
Cherry picked into 3.0.3

**Pull request merged**